src/constants/license.js

```diff
@@ -1,22 +1,45 @@
-export const CC_LICENSES = [
+export const CC_LICENSES = /** @type {const} */ ([
   'by',
   'by-sa',
   'by-nd',
   'by-nc',
   'by-nc-sa',
   'by-nc-nd',
-]
+])
+
+/**
+ * @template T
+ * @typedef {T extends `${infer P}-${infer PP}-${infer PPP}` ? [P, PP, PPP] : T extends `${infer P}-${infer PP}` ? [P, PP] : [T]} PartitionLicense
+ */
+
+/** @typedef {PartitionLicense<CCLicense>[number]} LicensePart */
```

Comment:
This is just me being kind of silly with TS string types. We could just as easily (and probably should just) use a type directly as `'by' | 'sa' | 'nd' | 'nc'` :stuck_out_tongue_closed_eyes:

Reply:
I tend to agree with your comment 😆  I mean it looks very interesting, but it would probably take me years to understand what it means :)